.circleci/config.yml

```diff
@@ -0,0 +1,60 @@
+version: 2.1
+
+aliases:
+- &restore_cache
+  restore_cache:
+    key: go-mod-v1-{{ checksum "go.sum" }}
+- &save_cache
+  save_cache:
+    key: go-mod-v1-{{ checksum "go.sum" }}
+    paths:
+    - "/go/pkg/mod"
+
+commands:
+  install-golangci-lint:
+    description: Install golangci-lint
+    parameters:
+      version:
+        type: string
+        default: 1.23.6
```

Comment:
We can use 1.31 as we are starting a fresh project.

Reply:
There is also this: https://hub.docker.com/r/golangci/golangci-lint/tags as an alternative to installing it.

Reply:
They recommend installing the release binary: https://golangci-lint.run/usage/install/#other-ci.

Reply:
But it can be locked to a specific release so it is still reproducible:  `image: golangci/golangci-lint:v1.25.1`
And the docker image does include the release binary.

Reply:
Yes, completely understand. I was just saying that they recommend using the release binary from github as a CI method. I am 0/5 on either.

Reply:
I personally would prefer docker, due to the code being more readable IMO.